go/service/gregor.go

```diff
@@ -86,14 +90,60 @@ func (g *gregorHandler) ShouldRetryOnConnect(err error) bool {
 
 func (g *gregorHandler) BroadcastMessage(ctx context.Context, m gregor1.Message) error {
 	g.G().Log.Debug("gregor handler: broadcast: %+v", m)
-	return nil
+
+	obm := m.ToOutOfBandMessage()
+	if obm == nil {
+		return fmt.Errorf("gregor handler: out-of-band message nil")
+	}
+
+	if obm.System() == nil {
+		return errors.New("nil system in out of band message")
+	}
+
+	switch obm.System().String() {
+	case "kbfs.favorites":
+		return g.kbfsFavorites(ctx, obm)
+	default:
+		return fmt.Errorf("unhandled system: %s", obm.System())
+	}
 }
 
 func (g *gregorHandler) Shutdown() {
 	g.conn.Shutdown()
 	g.conn = nil
 }
 
+func (g *gregorHandler) kbfsFavorites(ctx context.Context, m gregor.OutOfBandMessage) error {
+	if m.Body() == nil {
+		return errors.New("gregor handler for kbfs.favorites: nil message body")
+	}
+	body, err := jsonw.Unmarshal(m.Body().Bytes())
```

Comment:
I would, but didn't know what the object would be yet, so just used jsonw for now.